src/ol/interaction/extentinteraction.js

```diff
@@ -0,0 +1,460 @@
+goog.provide('ol.interaction.Extent');
+goog.provide('ol.interaction.ExtentEvent');
+goog.provide('ol.interaction.ExtentEventType');
+
+goog.require('ol');
+goog.require('ol.Feature');
+goog.require('ol.MapBrowserEvent.EventType');
+goog.require('ol.MapBrowserPointerEvent');
+goog.require('ol.coordinate');
+goog.require('ol.events');
+goog.require('ol.events.Event');
+goog.require('ol.extent');
+goog.require('ol.geom.GeometryLayout');
+goog.require('ol.geom.GeometryType');
+goog.require('ol.geom.Point');
+goog.require('ol.geom.Polygon');
+goog.require('ol.interaction.Pointer');
+goog.require('ol.layer.Vector');
+goog.require('ol.source.Vector');
+
+
+/**
+ * @enum {string}
+ */
+ol.interaction.ExtentEventType = {
+  /**
+   * Triggered after the extent is changed
+   * @event ol.interaction.ExtentEvent
+   * @api
+   */
+  EXTENTCHANGED: 'extentchanged'
+};
+/**
+ * @classdesc
+ * Events emitted by {@link ol.interaction.ExtentInteraction} instances are instances of
+ * this type.
+ *
+ * @constructor
+ * @param {ol.Extent} extent the new extent
+ * @extends {ol.events.Event}
+ */
+ol.interaction.ExtentEvent = function(extent) {
+  goog.base(this, ol.interaction.ExtentEventType.EXTENTCHANGED);
+};
+goog.inherits(ol.interaction.ExtentEvent, ol.events.Event);
+
+/**
+ * @classdesc
+ * Allows the user to draw a vector box by clicking and dragging on the map.
+ * Once drawn, the vector box can be modified by draging its vertices or edges.
+ * This interaction is only supported for mouse devices.
+ *
+ * @constructor
+ * @extends {ol.interaction.Pointer}
+ * @fires ol.ExtentEvent
+ * @param {olx.interaction.ExtentOptions} opts Options.
+ * @api
+ */
+ol.interaction.Extent = function(opts) {
+  /**
+   * Extent Extent
+   * @type {ol.Extent}
+   * @private
+   */
+  this.extent = null;
+  this.pointerHandler = null;
+
+  this.pixelTolerance_ = 10;
+  this.lastPixel_ = null;
```

Comment:
You should add `@type` (and `@private`, if appropriate) annotations for any member properties.